logbook-json/src/main/java/org/zalando/logbook/json/JsonPathBodyFilters.java

```diff
@@ -0,0 +1,91 @@
+package org.zalando.logbook.json;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.zalando.logbook.BodyFilter;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class JsonPathBodyFilters {
+
+    private static final Configuration conf = Configuration.builder().options(Option.AS_PATH_LIST).build();
+
+    @RequiredArgsConstructor
+    public static class JsonPathBodyFilterBuilder {
+        private final JsonPath jsonPath;
+
+        public BodyFilter delete() {
+            return (JsonBodyFilter) documentContext -> documentContext.delete(jsonPath).jsonString();
+        }
+
+        public BodyFilter replace(String replacement) {
+            return replaceInternal(replacement);
+        }
+
+        public BodyFilter replace(Number replacement) {
+            return replaceInternal(replacement);
+        }
+
+        public BodyFilter replace(Boolean replacement) {
+            return replaceInternal(replacement);
+        }
+
+        public BodyFilter replace(String matchRegex, String replacementRegex) {
+            Pattern pattern = Pattern.compile(matchRegex);
+            return (JsonBodyFilter) documentContext -> {
+                List<String> paths = documentContext.read(jsonPath);
+                Object jsonObject = documentContext.json();
+                paths.forEach(path -> {
+                    Object obj = JsonPath.compile(path).read(jsonObject);
+                    replaceWithRegex(obj, pattern, replacementRegex)
+                            .ifPresent(s -> documentContext.set(path, s));
+                });
+                return documentContext.jsonString();
+            };
+        }
+
+        private BodyFilter replaceInternal(Object replacement) {
+            return (JsonBodyFilter) documentContext -> {
+                List<String> obj = documentContext.read(jsonPath);
+                obj.forEach(path -> documentContext.set(path, replacement));
+                documentContext.set(jsonPath, replacement);
```

Comment:
When replacing with a static replacement, I guess you don't need to read first, you should be able to call `set` directly:

```suggestion
                documentContext.set(jsonPath, replacement);
```

Reply:
If we do so, we'll lose all the benefits from `AS_PATH_LIST`...